python/ray/data/datasource/file_based_datasource.py

```diff
@@ -355,16 +388,21 @@ def _write_block(
             "Subclasses of FileBasedDatasource must implement _write_files()."
         )
 
-    def _file_format(self):
-        """Returns the file format string, to be used as the file extension
+    @staticmethod
+    def _file_extension() -> str:
+        """Returns the file extension string, to be used as the file extension
         when writing files.
 
         This method should be implemented by subclasses.
         """
         raise NotImplementedError(
-            "Subclasses of FileBasedDatasource must implement _file_format()."
+            "Subclasses of FileBasedDatasource must implement _file_extension()."
         )
 
+    @classmethod
+    def file_extension_filter(cls) -> PathPartitionFilter:
+        return FileExtensionFilter(cls._file_extension())
```

Comment:
Nit: What's the motivation behind defining this abstraction? I'm not sure if reduces much complexity.

Pros of `file_extension_filter`:
* Makes read API constructor signatures less verbose

```
partition_filter: PathPartitionFilter = (
        ParquetBaseDatasource.file_extension_filter()
    ),
```

```
partition_filter: PathPartitionFilter = (
        FileExtensionFilter(ParquetBaseDatasource.file_extension())

    ),
```

Cons:
* Adds a layer of indirection.
* Complicates `FileBasedDatasource` implementation.